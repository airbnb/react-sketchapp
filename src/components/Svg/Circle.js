import React from 'react';
import { pathProps, numberProp } from './props';

module.exports = class Circle extends React.Component {
  static propTypes = {
    ...pathProps,
    cx: numberProp.isRequired,
    cy: numberProp.isRequired,
    r: numberProp.isRequired,
  };

  static defaultProps = {
    cx: 0,
    cy: 0,
    r: 0,
  };

  render() {
    const { children, ...rest } = this.props;
    return <svg_circle {...rest}>{children}</svg_circle>;
  }
};
