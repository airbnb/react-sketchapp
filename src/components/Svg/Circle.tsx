import * as React from 'react';
import { pathProps, numberProp } from './props';
import * as PropTypes from 'prop-types';

const propTypes = {
  ...pathProps,
  cx: numberProp.isRequired,
  cy: numberProp.isRequired,
  r: numberProp.isRequired,
};

type Props = PropTypes.InferProps<typeof propTypes>;

export class Circle extends React.Component<Props> {
  static propTypes = propTypes;

  static defaultProps = {
    cx: 0,
    cy: 0,
    r: 0,
  };

  render() {
    const { children, ...rest } = this.props;
    return <svg_circle {...rest}>{children}</svg_circle>;
  }
}
