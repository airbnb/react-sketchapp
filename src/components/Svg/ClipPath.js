import React from 'react';
import PropTypes from 'prop-types';

module.exports = class ClipPath extends React.Component {
  static propTypes = {
    id: PropTypes.string.isRequired,
    children: PropTypes.node,
  };

  render() {
    return (
      <svg_clipPath id={this.props.id}>{this.props.children}</svg_clipPath>
    );
  }
};
