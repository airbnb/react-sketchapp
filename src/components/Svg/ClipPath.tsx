import * as React from 'react';
import * as PropTypes from 'prop-types';

const propTypes = {
  id: PropTypes.string.isRequired,
  children: PropTypes.node,
};

type Props = PropTypes.InferProps<typeof propTypes>;

export class ClipPath extends React.Component<Props> {
  static propTypes = propTypes;

  render() {
    return <svg_clipPath id={this.props.id}>{this.props.children}</svg_clipPath>;
  }
}
