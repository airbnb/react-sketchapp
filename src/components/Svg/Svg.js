import React from 'react';
import PropTypes from 'prop-types';
import View from '../View';

module.exports = class Svg extends React.Component {
  static displayName = 'Svg';
  static propTypes = {
    ...View.propTypes,
    opacity: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    // more detail https://svgwg.org/svg2-draft/coords.html#ViewBoxAttribute
    viewBox: PropTypes.string,
    preserveAspectRatio: PropTypes.string,
  };

  static defaultProps = {
    preserveAspectRatio: 'xMidYMid meet',
  };

  render() {
    const { children, ...rest } = this.props;

    return <svg {...rest}>{children}</svg>;
  }
};
