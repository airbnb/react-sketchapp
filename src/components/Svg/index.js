module.exports = require('./Svg');

module.exports.Svg = require('./Svg');
module.exports.Circle = require('./Circle');
module.exports.ClipPath = require('./ClipPath');
module.exports.Defs = require('./Defs');
module.exports.Ellipse = require('./Ellipse');
module.exports.G = require('./G');
module.exports.Image = require('./Image');
module.exports.Line = require('./Line');
module.exports.LinearGradient = require('./LinearGradient');
module.exports.Path = require('./Path');
module.exports.Pattern = require('./Pattern');
module.exports.Polygon = require('./Polygon');
module.exports.Polyline = require('./Polyline');
module.exports.RadialGradient = require('./RadialGradient');
module.exports.Rect = require('./Rect');
module.exports.Stop = require('./Stop');
module.exports.Symbol = require('./Symbol');
module.exports.Text = require('./Text');
module.exports.TextPath = require('./TextPath');
module.exports.TSpan = require('./TSpan');
module.exports.Use = require('./Use');
