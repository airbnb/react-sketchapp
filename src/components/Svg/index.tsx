export { Circle } from './Circle';
export { ClipPath } from './ClipPath';
export { Defs } from './Defs';
export { Ellipse } from './Ellipse';
export { G } from './G';
export { SVGImage as Image } from './Image';
export { Line } from './Line';
export { LinearGradient } from './LinearGradient';
export { Path } from './Path';
export { Pattern } from './Pattern';
export { Polygon } from './Polygon';
export { Polyline } from './Polyline';
export { RadialGradient } from './RadialGradient';
export { Rect } from './Rect';
export { Stop } from './Stop';
export { Symbol } from './Symbol';
export { Text } from './Text';
export { TextPath } from './TextPath';
export { TSpan } from './TSpan';
export { Use } from './Use';
export { Svg as default } from './Svg';
