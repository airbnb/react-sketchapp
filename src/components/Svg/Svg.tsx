import * as React from 'react';
import * as PropTypes from 'prop-types';
import { ViewPropTypes } from '../View';
import { StyleSheet } from '../../stylesheet';
import { Circle } from './Circle';
import { ClipPath } from './ClipPath';
import { Defs } from './Defs';
import { Ellipse } from './Ellipse';
import { G } from './G';
import { SVGImage as Image } from './Image';
import { Line } from './Line';
import { LinearGradient } from './LinearGradient';
import { Path } from './Path';
import { Pattern } from './Pattern';
import { Polygon } from './Polygon';
import { Polyline } from './Polyline';
import { RadialGradient } from './RadialGradient';
import { Rect } from './Rect';
import { Stop } from './Stop';
import { Symbol } from './Symbol';
import { Text } from './Text';
import { TextPath } from './TextPath';
import { TSpan } from './TSpan';
import { Use } from './Use';

const propTypes = {
  ...ViewPropTypes,
  opacity: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  // more detail https://svgwg.org/svg2-draft/coords.html#ViewBoxAttribute
  viewBox: PropTypes.string,
  preserveAspectRatio: PropTypes.string,
  xmlns: PropTypes.string,
  'xmlns:xlink': PropTypes.string,
};

export type Props = PropTypes.InferProps<typeof propTypes>;

export class Svg extends React.Component<Props> {
  static displayName = 'Svg';

  static propTypes = propTypes;

  static defaultProps = {
    preserveAspectRatio: 'xMidYMid meet',
  };

  static Circle = Circle;

  static ClipPath = ClipPath;

  static Defs = Defs;

  static Ellipse = Ellipse;

  static G = G;

  static Image = Image;

  static Line = Line;

  static LinearGradient = LinearGradient;

  static Path = Path;

  static Pattern = Pattern;

  static Polygon = Polygon;

  static Polyline = Polyline;

  static RadialGradient = RadialGradient;

  static Rect = Rect;

  static Stop = Stop;

  static Symbol = Symbol;

  static Text = Text;

  static TextPath = TextPath;

  static TSpan = TSpan;

  static Use = Use;

  render() {
    const { children, style, ...rest } = this.props;

    return (
      <sketch_svg {...rest} style={StyleSheet.flatten(style)}>
        {children}
      </sketch_svg>
    );
  }
}
