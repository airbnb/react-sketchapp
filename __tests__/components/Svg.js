import React from 'react';
import renderer from 'react-test-renderer';
import Svg from '../../src/components/Svg';

describe('<Svg />', () => {
  it('passes its children', () => {
    const tree = renderer
      .create(
        <Svg
          xmlns="http://www.w3.org/2000/svg"
          width="494"
          height="447"
          viewBox="0 0 494 447"
        >
          <Svg.G fill="none" fillRule="evenodd">
            <Svg.Path
              fill="#FFAE00"
              d="M247 447L0 160 107 15 247 0l140 15 107 145"
            />
            <Svg.Path fill="#EC6C00" d="M247 447L0 160h494" />
            <Svg.Path fill="#FFAE00" d="M247 447L100 160h294" />
            <Svg.Path fill="#FFEFB4" d="M247 0L100 160h294" />
            <Svg.Path
              fill="#FFAE00"
              d="M107 15L52 88 0 160h101M387 15l55 73 52 72H393"
            />
            <Svg.Path
              fill="#FED305"
              d="M107 15l-7 145L247 0m140 15l7 145L247 0"
            />
          </Svg.G>
        </Svg>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
