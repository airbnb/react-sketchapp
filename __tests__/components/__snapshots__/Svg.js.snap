exports[`<Svg /> passes its children 1`] = `
<svg
  height="447"
  preserveAspectRatio="xMidYMid meet"
  viewBox="0 0 494 447"
  width="494"
  xmlns="http://www.w3.org/2000/svg">
  <svg_g
    fill="none"
    fillRule="evenodd">
    <svg_path
      d="M247 447L0 160 107 15 247 0l140 15 107 145"
      fill="#FFAE00" />
    <svg_path
      d="M247 447L0 160h494"
      fill="#EC6C00" />
    <svg_path
      d="M247 447L100 160h294"
      fill="#FFAE00" />
    <svg_path
      d="M247 0L100 160h294"
      fill="#FFEFB4" />
    <svg_path
      d="M107 15L52 88 0 160h101M387 15l55 73 52 72H393"
      fill="#FFAE00" />
    <svg_path
      d="M107 15l-7 145L247 0m140 15l7 145L247 0"
      fill="#FED305" />
  </svg_g>
</svg>
`;
